vent/coordinator/ipc.py

```diff
@@ -1,28 +1,90 @@
 import threading
 import time
+from enum import Enum, auto
+import pickle
+import socket
+
+buffer_size = 65536
+
+
+class IPCCommand(Enum):
+    START = auto()
+    STOP = auto()
+    GET_SENSORS = auto()
+    GET_CONTROL = auto()
+    SET_CONTROL = auto()
+    GET_ACTIVE_ALARMS = auto()
+    GET_LOGGED_ALARMS = auto()
+    CLEAR_LOGGED_ALARMS = auto()
+
+
+class IPCMessage:
+    def __init__(self, command, args=None):
+        """
+        :param command: ENUM in IPCCommand
+        """
+        self.command = command
+        # TODO: current only assume one arg
+        self.args = args
 
 
 class IPC:
     # Class for communicating between processes either by sockets
     # or named-pipes
     #   Functions:
-    def __init__(self, listen=False):
+    def __init__(self, listen, addr=None, port=None):
+        self.listen = listen
+        self.s = socket.socket()
```

Comment:
socket needs to have `setblocking(False)` or else all these calls block, which will make upstream code really hard to use.

Reply:
Changing to RPC solves this issue.